File: utils/cipher.go
package utils

import (
	"math/big"

	"github.com/iden3/go-iden3-crypto/babyjub"
)

func GenerateKeyPair() (babyjub.PrivateKey, *babyjub.PublicKey) {
	privkey := babyjub.NewRandPrivKey()
	return privkey, privkey.Public()
}

func Encrypt(message *big.Int, publicKey *babyjub.PublicKey, k *big.Int) (*babyjub.Point, *babyjub.Point) {
	// c1 = [k] * G
	c1 := babyjub.NewPoint().Mul(k, babyjub.B8)
	// s = [k] * publicKey
	s := babyjub.NewPoint().Mul(k, publicKey.Point())
	// m = [message] * G
	m := babyjub.NewPoint().Mul(message, babyjub.B8)
	// c2 = m + s
	c2p := babyjub.NewPointProjective().Add(m.Projective(), s.Projective())
	return c1, c2p.Affine()
}

File: ts/src/cipher.ts
// import { Base8, mulPointEscalar, Point, addPoint } from "@zk-kit/baby-jubjub"

import * as pkg from "@iden3/js-crypto";
const { babyJub } = pkg;

export function Encrypt(message : bigint, pubKey : [bigint,bigint], k : bigint) : [[bigint,bigint], [bigint,bigint]] {
    // c1 = k * G
    const c1 = babyJub.mulPointEscalar(babyJub.Base8, k);
    // s = k * pubKey
    const s = babyJub.mulPointEscalar(pubKey, k);
    // m = message * G
    const m = babyJub.mulPointEscalar(babyJub.Base8, message);
    // c2 = m + s
    const c2 = babyJub.addPoint(m, s);
    return [c1, c2];
}